components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/sync/WorkManagerSyncManager.kt

```diff
@@ -299,9 +301,24 @@ class WorkManagerSyncWorker(
             // Short-circuit if there are no configured stores.
             // Don't update the "last-synced" timestamp because we haven't actually synced anything.
             return Result.success()
-        }.also { stores ->
+        }
+
+        val passwordStore = syncableStores.entries.find { it.key == SyncEngine.Passwords }?.value
+
+        // If password storage is configured to be synced, we need to make sure it's decrypted during a sync.
+        passwordStore?.let {
+            val keySharedPreferences = SecureAbove22Preferences(context)
+
+            (it as LockableStore).ensureUnlocked(
+                keySharedPreferences.getString("passwords_key") ?: throw IllegalStateException(
+                    "'passwords_key' encryption key must be set if password sync is enabled"
+                )
+            )
+        }
```

Comment:
nit: Again my `let` overuse nit :)

I feel like this would be much more readable with just an ` if`. However here I wonder if we could just use something like:

```kotlin
passwordStore?.ensureUnlocked(
    SecureAbove22Preferences(context).getString("passwords_key")
    ?: ....
)
```